### PR TITLE
Add Ecosystem Sustainability Program (ESP) Page

### PR DESCRIPTION
**Pull request by bensternthal**
This PR adds information about the OpenJS Ecosystem Sustainability Program, including guidelines and best practices for participating with our first partner, HeroDevs. For additional background, please see #1277 

@ljharb @joesepi @tobie 

I'd like to fast-track this PR.

### CODE REVIEW

**Review by ljharb — APPROVED** [review]

**Comment by ljharb** [comment]
(separate comment for fast track thumbs-upping)

File: README.md
Comment (tobie) [inline README.md:248]:
```suggestion
* [OpenJS Ecosystem Sustainability Program](./project-resources/ESP/ECOSYSTEM_SUSTAINABILITY_PROGRAM.md) - The Ecosystem Sustainability Program is an opt-in OpenJS partner program to help secure outdated software and create new revenue streams for our projects.  
```
Creating revenue streams for commercial participants shouldn’t be one of our goals, imho.

File: project-resources/ESP/ECOSYSTEM_SUSTAINABILITY_PROGRAM.md
Comment (tobie) [inline project-resources/ESP/ECOSYSTEM_SUSTAINABILITY_PROGRAM.md:5]:
```suggestion
The Ecosystem Sustainability Program is an opt-in OpenJS partner program to help secure outdated software and create new revenue streams for our projects.  
```

Same.

File: project-resources/ESP/ECOSYSTEM_SUSTAINABILITY_PROGRAM.md
Comment (tobie) [inline project-resources/ESP/ECOSYSTEM_SUSTAINABILITY_PROGRAM.md:24]:
```suggestion
- A partner must be providing support end-of-life (EOL) versions of the project
```

alt: “no longer supported”

File: project-resources/ESP/ECOSYSTEM_SUSTAINABILITY_PROGRAM.md
Comment (tobie) [inline project-resources/ESP/ECOSYSTEM_SUSTAINABILITY_PROGRAM.md:25]: How does the “requirements will vary per partner” work in practice? It feels like this should be negotiated by the foundation for all of the projects and partners and not the fruit of 1:1 negotiations between projects and partners.

File: project-resources/ESP/ECOSYSTEM_SUSTAINABILITY_PROGRAM.md
Comment (tobie) [inline project-resources/ESP/ECOSYSTEM_SUSTAINABILITY_PROGRAM.md:36]:
```suggestion
In most cases, projects do not have bank accounts and must [sign up](https://opencollective.com/opensource/apply) with Open Source Collective to be a fiscal host. This means Open Collective will hold the funds on your behalf, handle taxes, and enable you to pay vendors and contractors. 
```

File: project-resources/ESP/ECOSYSTEM_SUSTAINABILITY_PROGRAM.md
Comment (tobie) [inline project-resources/ESP/ECOSYSTEM_SUSTAINABILITY_PROGRAM.md:41]: I have no idea what the LF G&A fee means

File: project-resources/ESP/ECOSYSTEM_SUSTAINABILITY_PROGRAM.md
Comment (tobie) [reply project-resources/ESP/ECOSYSTEM_SUSTAINABILITY_PROGRAM.md:25]: Also worth linking to the section below where more info is provided.

**Review by tobie — COMMENTED** [review]
I’m a little uneasy about the idea of partner-specific guidelines. What’s the rationale, here?

File: README.md
Comment (ljharb) [reply README.md:248]: Without it, there won't be any new revenue streams for our projects, so i'm not sure how they can be separated.

File: project-resources/ESP/ECOSYSTEM_SUSTAINABILITY_PROGRAM.md
Comment (ljharb) [reply project-resources/ESP/ECOSYSTEM_SUSTAINABILITY_PROGRAM.md:41]: https://cdn.platform.linuxfoundation.org/agreements/pqca.pdf

> A General & Administrative (G&A) fee will be applied by The Linux Foundation to funds raised to cover membership records, finance, accounting, and human resources operations. The G&A fee will be 9% of the Directed Fund’s first $1,000,000 of gross receipts each year and 6% of the Directed Fund’s gross receipts each year over $1,000,000.

Comment (tobie) [reply project-resources/ESP/ECOSYSTEM_SUSTAINABILITY_PROGRAM.md:41]: Is this something the OpenJS is subject to? If so, let’s spell acronym out and link it.

File: README.md
Comment (tobie) [reply README.md:248]: My point is that it shouldn’t be a goal for the foundation. Let’s say that tomorrow a philanthropy shows up and wants to do a partnership where 100% of the revenue goes to the projects and the foundation. we shouldn’t care that there isn’t a revenue stream for them.

**Review by tobie — APPROVED** [review]
> I’m a little uneasy about the idea of partner-specific guidelines. What’s the rationale, here?

Discussed this offline with @bensternthal. There was agreement to make the guidelines more generic with just the referral links / brand being partner specific. LGTM once this and other comments addressed.

Understand the urgency to merge this. However, please ping CPC members before fast-tracking (eg through the private list or by tagging them here) as this is a bit out of what’s reasonable to fast track imho.

**Comment by bensternthal** [comment]
I'll post an update with the HeroDevs recommendations edited to be general recommendations. @tobie is right that these recommendations are not vendor specific. 

If there is a requirement to ping CPC members before fast-tracking it should be added to the [docs](https://github.com/openjs-foundation/cross-project-council/blob/main/governance/GOVERNANCE.md#fast-tracking-prs).  Right now it states that if folks disagree they should down vote or remove the label.

File: project-resources/ESP/ECOSYSTEM_SUSTAINABILITY_PROGRAM.md
Comment (ljharb) [reply project-resources/ESP/ECOSYSTEM_SUSTAINABILITY_PROGRAM.md:41]: Yes, as far as i'm aware. linking sounds good.

Comment (bensternthal) [reply project-resources/ESP/ECOSYSTEM_SUSTAINABILITY_PROGRAM.md]: Will add link when I find it :)

Comment (ljharb) [reply project-resources/ESP/ECOSYSTEM_SUSTAINABILITY_PROGRAM.md]: just linking to the PDF i linked above and referencing the section number should be fine

**Comment by ovflowd** [comment]
> > If there is a requirement to ping CPC members before fast-tracking it should be added to the [docs](https://github.com/openjs-foundation/cross-project-council/blob/main/governance/GOVERNANCE.md?rgh-link-date=2024-05-29T21%3A41%3A54Z#fast-tracking-prs). Right now it states that if folks disagree they should down vote or remove the label.
> 
> There is indeed no such requirements. Imho, this PR largely exceeds what we customarily accept for fast track. I know there’s a presentation later this week for which this is needed so it felt acceptable to compromise but would require making sure people get a chance to see before it gets merged by doing outreach. Removing the fast-track label for now. Happy to allow it back in if folks think the proposed compromise is acceptable.

Afaik there is a fast-tracking rule. @bensternthal this is fast-tracked, you can merge it. https://github.com/openjs-foundation/cross-project-council/blob/main/governance/GOVERNANCE.md?plain=1#L197

**Review by ovflowd — APPROVED** [review]

**Comment by tobie** [comment]
For the record, @ovflowd, I had removed the fast-track label (as per the line following the one you linked). I’m open to re-adding it if (1) this needs to be fast-tracked to meet the Friday deadline and (2) there’s additional outreach that’s done to make sure CPC members see it before it’s merged.

**Review by VoltrexKeyva — APPROVED** [review]

**Review by marco-ippolito — APPROVED** [review]

**Pull request merged**
